contracts/contracts/assets/rlp/src/main.sw

```diff
@@ -0,0 +1,262 @@
+const DEFAULT_SUB_ID: SubId = SubId::zero();
+
+storage {
+    gov: Identity = ZERO_IDENTITY,
+    is_initialized: bool = false,
+    
+    name: StorageString = StorageString {},
+    symbol: StorageString = StorageString {},
+
+    /// total supply of RLP
+    /// only really 1 SubId is utilized for RLP minting
+    total_supply: StorageMap<AssetId, u64> = StorageMap {},
+    /// value for this is ALWAYS 1
+    total_assets: u64 = 0,
+
+    approved_minters: StorageMap<Identity, bool> = StorageMap {},
+}
+
+impl RLP for Contract {
+    #[storage(read, write)]
+    fn initialize() {
+        require(
+            !storage.is_initialized.read(), 
+            Error::RLPAlreadyInitialized
+        );
+        storage.is_initialized.write(true);
+
+        storage.name.write_slice(String::from_ascii_str("RLP"));
+        storage.symbol.write_slice(String::from_ascii_str("RLP"));
+
+        let sender = get_sender();
+        log(SetNameEvent { 
+            asset: _get_id(),
+            name: storage.name.read_slice(),
+            sender
+        });
+        log(SetSymbolEvent { 
+            asset: _get_id(),
+            symbol: storage.symbol.read_slice(),
+            sender
+        });
+        log(SetDecimalsEvent { 
+            asset: _get_id(),
+            decimals: DECIMALS,
+            sender
+        });
+        
+        let sender = get_sender();
+        storage.gov.write(sender);
+        storage.approved_minters.insert(sender, true);
+        log(SetGov { gov: sender });
+        log(SetApprovedMinter { minter: sender, is_active: true });
+    }
+
+    /*
+          ____     _       _           _       
+         / / /    / \   __| |_ __ ___ (_)_ __  
+        / / /    / _ \ / _` | '_ ` _ \| | '_ \ 
+       / / /    / ___ \ (_| | | | | | | | | | |
+      /_/_/    /_/   \_\__,_|_| |_| |_|_|_| |_|                         
+    */
+    #[storage(read, write)]
+    fn set_gov(gov: Identity) {
+        _only_gov();
+        storage.gov.write(gov);
+        log(SetGov { gov });
+    }
+
+    #[storage(read, write)]
+    fn set_minter(minter: Identity, is_active: bool) {
+        _only_gov();
+
+        storage.approved_minters.insert(minter, is_active);
+        log(SetApprovedMinter { minter, is_active });
+    }
+
+    /*
+          ____ __     ___
+         / / / \ \   / (_) _____      __
+        / / /   \ \ / /| |/ _ \ \ /\ / /
+       / / /     \ V / | |  __/\ V  V /
+      /_/_/       \_/  |_|\___| \_/\_/
+    */
+    /// Returns the AssetId of the RLP token
+    fn get_id() -> AssetId {
+        _get_id()
+    }
+
+    /// Returns the total supply of the RLP asset
+    #[storage(read)]
+    fn total_rlp_supply() -> u64 {
+        storage.total_supply.get(_get_id()).try_read().unwrap_or(0)
+    }
+}
+
+// https://docs.fuel.network/docs/sway-standards/src-20-native-asset/
+impl SRC20 for Contract {
+    #[storage(read)]
+    fn name(_asset: AssetId) -> Option<String> {
+        Some(storage.name.read_slice().unwrap())
```

Comment:
IF we are using constants these of course no longer access storage. See decimals as an example.